### PR TITLE
Split the functionality of commit trigger to commit and PR triggers.

### LINKED ISSUE #814: Split the functionality of commit trigger to commit and PR triggers
We should split out the functionality what we have defined as a commit trigger today into 2 separate triggers. 1 for commits on a specific branch and another for all PRs.

Use Cases:
1) when a commit is tested in a PR, it is not at the same HEAD as master per say. So there might be some logic that is incompatible once merged, but passes in the feature branch.
2) I imagine it would be nice to have a source of truth history of your master branch when it comes to tests. This would help with regressions.
3) I believe there are lanes people want to run on commits to a branch rather than all PRs and commit. For example a slack notification.

### PR DESCRIPTION
**Pull request by AndrewMcBurney**
Fixes #814.

- [x] I have run `rspec` and corrected all errors
- [x] I have run `rubocop` and corrected all errors
- [x] I have run `npm run test` and corrected all errors
- [x] I have tested this change locally and tried to launch the server as well as access a project, and with that at least one build
- [x] If there is an existing issue, make sure to add `Fixes ...` as part of the PR body to reference the issue you're solving
- [x] Refactor the `check_for_new_commits_on_branches` method and add specs for the `CheckForNewCommitsOnGithubWorker` worker

### CODE REVIEW

**Comment by taquitos** [comment]
@AndrewMcBurney this is cool, what more do you think we need to be able to 🚢 this?

File: app/shared/factories/trigger_factory.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        triggers_to_add << FastlaneCI::CommitJobTrigger.new(branch: branch)
+      when FastlaneCI::JobTrigger::TRIGGER_TYPE[:pull_request]
+        triggers_to_add << FastlaneCI::PullRequestJobTrigger.new(branch: branch)
+      when FastlaneCI::JobTrigger::TRIGGER_TYPE[:manual]
+        # Nothing to do here, manual trigger is added by default
+        logger.debug("Manual trigger selected - this is enabled by default")
+      when FastlaneCI::JobTrigger::TRIGGER_TYPE[:nightly]
+        triggers_to_add << FastlaneCI::NightlyJobTrigger.new(
+          branch: branch,
+          hour: params[:hour].to_i,
```
Comment (nakhbari) [inline app/shared/factories/trigger_factory.rb]: should we assert that these timing params exist? Or maybe default them to midnight?

File: app/workers/check_for_new_commits_on_github_worker.rb [hunk 2]
```diff
@@ -43,14 +51,22 @@ def check_for_new_commits
       # All the shas for builds we have run and dump it to a set so we can filter with it
       local_build_shas_set = builds.map(&:sha).to_set
 
-      # Get all commits from the open PRs
-      open_pull_requests = github_service.open_pull_requests(repo_full_name: repo_full_name)
+      # Get all branches for all commit triggers
+      branches = project.find_triggers_of_type(trigger_type: :commit).map(&:branch)
+
+      # Get all commits from the open PRs given a set of branches
+      open_pull_requests = github_service.open_pull_requests(
```
Comment (nakhbari) [inline app/workers/check_for_new_commits_on_github_worker.rb]: do we still need this, if the commit trigger is only aimed to be triggered on new commits on the target branch?

File: web/app/dashboard/add-project-dialog/add-project-dialog.component.ts [hunk 3]
```diff
@@ -15,12 +15,13 @@ export interface AddProjectDialogConfig {
...
 interface TriggerOption {
   viewValue: string;
-  value: 'commit'|'nightly';
+  value: 'commit' | 'pull_request' | 'nightly';
 }
 
 const TRIGGER_OPTIONS: TriggerOption[] = [
-  {viewValue: 'for every commit and PR', value: 'commit'},
-  {viewValue: 'nightly', value: 'nightly'},
+  { viewValue: 'for every commit', value: 'commit' },
```
Comment (nakhbari) [inline web/app/dashboard/add-project-dialog/add-project-dialog.component.ts]: i think this might break some of the web app tests

Comment (AndrewMcBurney) [reply web/app/dashboard/add-project-dialog/add-project-dialog.component.ts]: I fixed the frontend tests ✅.

File: app/workers/check_for_new_commits_on_github_worker.rb [hunk 4]
```diff
@@ -15,10 +15,22 @@ module FastlaneCI
...
+    #
+    # @return [WorkerScheduler]
     attr_reader :scheduler
+
+    # @return [GitHubService]
     attr_reader :github_service
 
+    # Instantiates a new `CheckForNewCommitsOnGithubWorker` object.
+    #
+    # @param [ProviderCredential] provider_credential: The credential needed to communicate with GitHub API.
```
Comment (KrauseFx) [inline app/workers/check_for_new_commits_on_github_worker.rb]: so many comments 😍

**Comment by KrauseFx** [comment]
@AndrewMcBurney let us know when you want a review on this 👍

**Comment by AndrewMcBurney** [comment]
Hey @KrauseFx, it's good for a review now 🙂.

File: app/services/config_data_sources/json_project_data_source.rb [hunk 5]
```diff
@@ -134,18 +133,18 @@ def job_triggers_from_hash_array(job_trigger_array: nil)
...
-        # currently only supports 3 triggers
-        job_trigger = nil
-        if type == FastlaneCI::JobTrigger::TRIGGER_TYPE[:commit]
-          job_trigger = CommitJobTrigger.from_json!(job_trigger_hash)
-        elsif type == FastlaneCI::JobTrigger::TRIGGER_TYPE[:nightly]
-          job_trigger = NightlyJobTrigger.from_json!(job_trigger_hash)
-        elsif type == FastlaneCI::JobTrigger::TRIGGER_TYPE[:manual]
-          job_trigger = ManualJobTrigger.from_json!(job_trigger_hash)
+        case type
+        when FastlaneCI::JobTrigger::TRIGGER_TYPE[:commit]
```
Comment (nakhbari) [inline app/services/config_data_sources/json_project_data_source.rb]: Should we pull this out into it's own method so we don't have to duplicate it and for readability?

File: app/shared/factories/trigger_factory.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require_relative "abstract_factory"
+
+module FastlaneCI
+  # TODO: Until we make a proper interface to attach JobTriggers to a Project, let's add a manual one for the
```
[the remark this answers is not in the document]
Comment (AndrewMcBurney) [reply app/shared/factories/trigger_factory.rb]: Ahh, you're right - the comment doesn't apply anymore, good catch! I'll remove it 👍

File: app/services/config_data_sources/json_project_data_source.rb (hunk 5 above)
Comment (AndrewMcBurney) [reply app/services/config_data_sources/json_project_data_source.rb]: Will do 👍

**Pull request merged**